airgun/entities/host_new.py

```diff
@@ -47,6 +48,43 @@ def get_details(self, entity_name, widget_names=None):
         view.read(widget_names=widget_names)
         return view.read(widget_names=widget_names)
 
+    def assign_role_to_hostgroup(self, entity_name):
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        view.overview.details.edit.click()
+        self.browser.switch_to_window(self.browser.window_handles[1])
+        self.browser.plugin.ensure_page_safe()
+        hostgrp_view = HostGroupCreateView(self.browser)
+        actions = [
+            hostgrp_view.ansible_roles,
+            hostgrp_view.ansible_roles.more_item,
+            hostgrp_view.ansible_roles.select_pages,
+        ]
+        [action.click() for action in actions]
+        assign_role = hostgrp_view.ansible_roles.available_role.read().split(". ")
+        if assign_role[1] == 'RedHatInsights.insights-client':
+            hostgrp_view.ansible_roles.available_role.click()
+        hostgrp_view.ansible_roles.submit.click()
+        self.browser.switch_to_window(self.browser.window_handles[0])
+        self.browser.close_window(self.browser.window_handles[1])
+
+    def remove_hostgroup_role(self, entity_name):
+        view = self.navigate_to(self, 'NewDetails', entity_name=entity_name)
+        view.wait_displayed()
+        self.browser.plugin.ensure_page_safe()
+        view.overview.details.edit.click()
+        self.browser.switch_to_window(self.browser.window_handles[1])
+        self.browser.plugin.ensure_page_safe()
+        hostgrp_view = HostGroupCreateView(self.browser)
+        hostgrp_view.ansible_roles.click()
+        assign_role = hostgrp_view.ansible_roles.assigned_role.read().split(". ")
+        if assign_role[1] == 'RedHatInsights.insights-client':
+            hostgrp_view.ansible_roles.assigned_role.click()
```

Comment:
@amolpati30 I unresolved this comment cause I'm still a bit confused. Could you please give some explanation here?

Reply:
I only tried with adding & removing single ansible role, the task is whether role is assigning and removing properly. But i will try with user input. Right now I thing single input is sufficient to check the action.

Reply:
Well if you think it's fine, then I'll leave it at that:)

Reply:
I agree with @jameerpathan111 here. This method should take user input on which ansible role to remove from hostgroup so that this can be reused somewhere else if needed since this is part of airgun. I would suggest to make this more flexible.